app/src/main/java/com/github/siela1915/bootcamp/IngredientCheckAdapter.java

```diff
@@ -0,0 +1,71 @@
+package com.github.siela1915.bootcamp;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.CheckBox;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class IngredientCheckAdapter extends RecyclerView.Adapter<IngredientCheckAdapter.ICViewHolder> {
+    private final List<String> items;
+    private final List<String> selectedItems = new ArrayList<>();
+    public IngredientCheckAdapter(List<String> items) {
+        this.items = items;
+    }
+
+    @NonNull
+    @Override
+    public ICViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.ingredient_item_view, parent, false);
+        return new ICViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ICViewHolder holder, int position) {
+        String item = items.get(position);
+        holder.textView.setText(item);
+        holder.checkBox.setChecked(selectedItems.contains(item));
+        holder.checkBox.setOnClickListener(v -> {
+            if (selectedItems.contains(item)) {
+                selectedItems.remove(item);
+            } else {
+                selectedItems.add(item);
+            }
+            holder.checkBox.setChecked(selectedItems.contains(item));
+        });
+        holder.itemView.setOnClickListener(v -> {
+            if (selectedItems.contains(item)) {
+                selectedItems.remove(item);
+            } else {
+                selectedItems.add(item);
+            }
+            holder.checkBox.setChecked(selectedItems.contains(item));
+        });

```

Comment:
It looks like both listeners update `selectedItems` with the same logic. Maybe it would be better to extract this functionality into a separate method and call it from both click listeners to avoid code duplication and improve readability of `onBindViewHolder`.